src/solidity-frontend/solidity_convert.cpp

```diff
@@ -1179,12 +1186,24 @@ bool solidity_convertert::get_expr(
   case SolidityGrammar::ExpressionT::MemberCallClass:
   {
     assert(expr.contains("expression"));
+    expr.dump(4, ' ', false, nlohmann::detail::error_handler_t::strict).c_str();
     const nlohmann::json &callee_expr_json = expr["expression"];
 
+    // Function symbol id is c:@C@referenced_function_contract_name@F@function_name#referenced_function_id
+    // Using referencedDeclaration will point us to the original declared function. This works even for inherited function and overrided functions. 
+  
+    const int caller_id = callee_expr_json["referencedDeclaration"].get<std::uint16_t>();
+    const nlohmann::json caller_expr_json = find_decl_ref(caller_id);
+    const int contract_id = caller_expr_json["scope"].get<std::uint16_t>();
+
+    std::string ref_contract_name;
+    get_contract_name(contract_id, ref_contract_name);
+
     // obtain type info from symbol table
     std::string name, id;
     name = callee_expr_json["memberName"];
-    id = name;
+    id = "c:@C@" + ref_contract_name + "@F@" + name + "#" + i2string(caller_id);
```

Comment:
Is it possible to obtain the `id` uniformly by `get_function_definition_name()`?

Reply:
Yes, I will make corresponding changes.